pennylane_lightning_gpu/lightning_gpu.py

```diff
@@ -232,27 +234,40 @@ def __init__(
 
             super().__init__(wires, shots=shots, r_dtype=r_dtype, c_dtype=c_dtype)
 
-            if mpi is False:
+            if not mpi:
+                self._mpi = False
                 self._num_local_wires = self.num_wires
                 self._gpu_state = _gpu_dtype(c_dtype)(self._num_local_wires)
                 self._batch_obs = batch_obs
             else:
+                self._mpi = True
                 self._mpi_init_helper(self.num_wires)
 
-                if log2_mpi_buf_counts > self._num_local_wires:
-                    w_msg = "MPI buffer size is over the size of local state vector."
-                    warn(
-                        w_msg,
-                        RuntimeWarning,
-                    )
+                if mpi_buf_size < 0:
+                    raise TypeError(f"Unsupported mpi_buf_size value: {mpi_buf_size}")
+
+                if not mpi_buf_size:
+                    if mpi_buf_size & (mpi_buf_size - 1):
+                        raise TypeError(
+                            f"Unsupported mpi_buf_size value: {mpi_buf_size}. mpi_buf_size should be power of 2."
+                        )
 
-                if log2_mpi_buf_counts < 0:
-                    raise TypeError(f"Unsupported log2_mpi_buf_counts value: {log2_mpi_buf_counts}")
+                if not mpi_buf_size:
+                    if c_dtype is np.complex64:
+                        sv_memsize = self._num_local_wires + 3
+                    else:
+                        sv_memsize = self._num_local_wires + 4
+                    if (mpi_buf_size.bit_length() - 1) > (self._num_local_wires - 20):
```

Comment:
Why `self._num_local_wires - 20` here?

Reply:
please check the docstring. now `mpi_buf_size` is in megabytes, if mpi_buf_size is set as 1, it represents 1<<20 bytes. That's why 20 is chosen here.

Reply:
Thanks Vincent! To make it more straight forward. A convert function (megabytes to bytes) is added in the update. Please check.